src/main/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RefreshOrchestrator.java

```diff
@@ -78,14 +81,20 @@ public void validate(Long jobId, UserRequest userRequest) {
             throw new BadRequestException("Invalid JobId request");
         }
 
-
         if (userRequest != null && nullCheckPredicate.test(userRequest.getUserIds())) {
-            //Extract and Validate received users List
+            // Extract and Validate received users List
             parseRequestService.validateUserRequest(userRequest);
             log.info("Validated userIds {}", userRequest.getUserIds());
         }
+
+        // Ensure only permitted service is invoking caseworker refresh API
+        final String serviceName = securityUtils.getServiceName();
+        if (!authorisedServices.contains(serviceName)) {
+            throw new BadRequestException("Invoking service is not permitted to call the Refresh API");
+        }
     }
 
+    @Async
```

Comment:
I assume the move is required to allow the validation code to run before the reponse is sent ?